app/controllers/post.js

```diff
@@ -1,16 +1,89 @@
 const userModel = require('../models/User')
 const bcrypt = require('bcrypt')
 const jwt = require('jsonwebtoken')
+const PostModel = require('../models/Post')
+const HANDLER = require('../utils/response-helper')
+const STATUS = require('../utils/status-codes')
+const helper = require('../utils/uploader')
 
 module.exports = {
-  create: function (req, res, next) {
-    userModel.create({ name: req.body.name, email: req.body.email, password: req.body.password }, function (err, result) {
-      if (err) {
-        next(err)
-      } else {
-        res.json({ status: 'success', message: 'User added successfully!!!', data: null })
+  create: async (req, res, next) => {
+    const post = new PostModel(req.body)
+    if (req.file) {
+      helper.mapToDb(req, post)
+    }
+    try {
+      await post.save()
+      res.status(STATUS.CREATED).json({ post })
+    } catch (error) {
+      HANDLER.handleError(res, error)
+    }
+  },
+  delete: async (req, res, next) => {
+    const { id } = req.params
+    console.log(id)
+    try {
+      const post = await PostModel.findByIdAndRemove(id)
+      if (!post) {
+        return res.status(STATUS.NOT_FOUND).json({ message: 'No post exists' })
       }
+      res.status(STATUS.OK).json({ post: post, message: 'Deleted the post' })
```

Comment:
you are missing a very imp part here. You who owns a post can only delete his post or if her is an admin then can delete the post/

Reply:
Oh yes, I will do it.